Quadriga/src/main/java/edu/asu/spring/quadriga/authentication/AuthenticationSuccessHandlerImpl.java

```diff
@@ -0,0 +1,29 @@
+package edu.asu.spring.quadriga.authentication;
+
+import java.io.IOException;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.springframework.security.core.Authentication;
+import org.springframework.security.web.authentication.AuthenticationSuccessHandler;
+import org.springframework.stereotype.Service;
+
+import edu.asu.spring.quadriga.service.QuadrigaUserDetails;
+
+@Service("authSuccessHandler")
+public class AuthenticationSuccessHandlerImpl implements AuthenticationSuccessHandler {
+
+	@Override
+	public void onAuthenticationSuccess(HttpServletRequest request, HttpServletResponse response,
+			Authentication authentication) throws IOException, ServletException {
+		Object obj = authentication.getPrincipal();
+		if (obj instanceof QuadrigaUserDetails) {
+			QuadrigaUserDetails quadrigaUserDetails = (QuadrigaUserDetails) obj;
+			String userName = quadrigaUserDetails.getUsername();
+			quadrigaUserDetails.setUsername(userName.toLowerCase());
+		}
+		response.sendRedirect("/quadriga/auth/welcome");
```

Comment:
You cannot hardcode the url as the webapp might not always be called "quadriga". On the review instance for example, logging in leads to a page not found error.